lib/lightning_web/live/credential_live/oauth_component.ex

```diff
@@ -499,44 +666,65 @@ defmodule LightningWeb.CredentialLive.OauthComponent do
     end)
     |> assign_new(:authorize_url, fn %{client: client} ->
       if client do
-        socket.assigns.provider.authorize_url(
+        {_, default_scopes} =
+          LightningWeb.CredentialLive.Scopes.get_scopes(socket.assigns.provider)
+
+        module_name(socket.assigns.provider).authorize_url(
           client,
           build_state(socket.id, __MODULE__, socket.assigns.id),
-          []
+          default_scopes
         )
       end
     end)
   end
 
+  defp module_name(provider) do
+    case provider do
+      "google" -> Lightning.AuthProviders.Google
+      "salesforce" -> Lightning.AuthProviders.Salesforce
+    end
+  end
+
   defp handle_code_update(code, socket) do
     client = socket.assigns.client
 
     # NOTE: there can be _no_ refresh token if something went wrong like if the
     # previous auth didn't receive a refresh_token
 
-    case socket.assigns.provider.get_token(client, code: code) do
+    case module_name(socket.assigns.provider).get_token(client, code: code) do
       {:ok, client} ->
         client.token
         |> token_to_params()
         |> socket.assigns.update_body.()
 
-        {:ok, socket |> assign(authorizing: false, client: client)}
+        send_update(LightningWeb.CredentialLive.FormComponent,
+          id: socket.assigns.parent_id,
+          authorization_status: :success
+        )
```

Comment:
Instead of sending the parent_id, and creating a dependency on the upstream component - would it not be better to pass in a function like we do with `update_body` already?